lib/esm/ModuleChunkLoadingRuntimeModule.js

```diff
@@ -224,19 +232,36 @@ class ModuleChunkLoadingRuntimeModule extends RuntimeModule {
 													: `if(${hasJsMatcher("chunkId")}) {`,
 												Template.indent([
 													"// setup Promise in chunk cache",
-													`var promise = ${importFunctionName}(${
-														compilation.outputOptions.publicPath === "auto"
-															? JSON.stringify(rootOutputDir)
-															: RuntimeGlobals.publicPath
-													} + ${
-														RuntimeGlobals.getChunkScriptFilename
-													}(chunkId)).then(installChunk, ${runtimeTemplate.basicFunction(
-														"e",
-														[
-															"if(installedChunks[chunkId] !== 0) installedChunks[chunkId] = undefined;",
-															"throw e;"
-														]
-													)});`,
+													outputModule && isESMLibrary
+														? // Use new URL() for ESM library output
+															`var promise = ${importFunctionName}(new URL(${JSON.stringify(
+																rootOutputDir
+															)} + ${
+																RuntimeGlobals.getChunkScriptFilename
```

Comment:
They are still not statically analyzable... we need more complex logic here, currently we generate more code, so I think we should solve `new URL()` and filenames for chunks together to avoid extra code generation